src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp

```diff
@@ -0,0 +1,149 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+#include <OpenSpaceToolkit/Core/Types/Index.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace dynamics
+{
+
+using ostk::core::types::Index;
+
+using ostk::math::object::VectorXd;
+
+Tabulated::Tabulated(
+    const Array<Instant>& anInstantArray,
+    const MatrixXd& aContributionProfile,
+    const Array<Shared<const CoordinatesSubset>>& aWriteCoordinatesSubsets
+)
+    : Dynamics("Tabulated"),
+      contributionProfile_(aContributionProfile),
+      instants_(anInstantArray),
+      writeCoordinatesSubsets_(aWriteCoordinatesSubsets)
+{
+    if (anInstantArray.getSize() != (Index)aContributionProfile.rows())
+    {
+        throw ostk::core::error::RuntimeError(
+            "Contribution profile must have the same number of rows as the number of instants."
+        );
+    }
+
+    if (aWriteCoordinatesSubsets
+            .map<Index>(
+                [](const auto& coordinatesSubset)
+                {
+                    return coordinatesSubset->getSize();
+                }
+            )
+            .reduce(std::plus<Index>()) != (Index)aContributionProfile.cols())
+    {
+        throw ostk::core::error::RuntimeError(
+            "Contribution profile must have the same number of columns as the sum of the sizes of the write "
+            "coordinates subsets."
+        );
+    }
+
+    VectorXd timestamps(anInstantArray.getSize());
+    Index i = 0;
+    for (const auto& instant : anInstantArray)
+    {
+        timestamps(i) = (instant - Instant::J2000()).inSeconds();
+        ++i;
+    }
+
+    interpolators_.reserve(aContributionProfile.cols());
+    for (i = 0; i < (Index)aContributionProfile.cols(); ++i)
+    {
+        // TBI: In the future we can allow the user to specify the interpolator type.
+        interpolators_.add(BarycentricRational(timestamps, aContributionProfile.col(i)));
```

Comment:
This is already checked in the interpolator. I don't think it's worth having another set of tests, it will raise an appropriate error if the instants are not sorted. But great point 💯 I'll add that to the docs.